src/pybel/io/mongodb.py

```diff
@@ -0,0 +1,204 @@
+    :return: the collection that now stores the graph
+    """
+    if client is None:
+        client = MongoClient()
+    if db_name is None:
+        db_name = DB_NAME
+    # Access (or create) the specified database
+    db = client[db_name]
+    # If no collection_name was passed
+    if collection_name is None:
+        # Attempt to create collections until a unique name is found
+        collection_name = _default_collection_name(db)
+    # Access (or create) the specified collection
+    collection = db[collection_name]
+    # If a collection with the same name already exists, drop its contents
+    collection.drop()
+
+    # Store all the nodes/edges in a list to be bulk added
+    documents = []
+
+    # Add the nodes to documents
+    for node in graph:
+        if not is_valid_node(node):
+            logger.warning(f'Invalid node encountered: {node}')
+        # Add a 'type' parameter to avoid confusing nodes and links
+        n = deepcopy(node)
+        n[TYPE] = NODE
+        documents.append(n)
+
+    # Add the edges to documents
+    edges = to_sbel(graph, yield_metadata=False)
+    for edge in edges:
+        if not is_valid_edge(edge):
+            logger.warning(f'Invalid edge encountered: {edge}')
+        # Add a 'type' parameter to avoid confusing nodes and links
+        e = deepcopy(edge)
+        e[TYPE] = LINK
+        documents.append(e)
+    # Insert the documents into the collection
+    collection.insert_many(documents)
+    return collection
+
+
+def find_nodes(
+    collection: Collection,
+    name: str = None,
+    identifier: str = None,
+    function: str = None,
+    variants: List[Variant] = None,
+) -> List[Mapping[str, Any]]:
+    """Find all the nodes that match the given criteria from a MongoDB Collection where a graph is stored.
+
+    :param collection: A MongoDB collection within a database where a PyBEL graph has been stored
+    :param name: The name of the desired node
+    :param identifier: The identifier of the desired node
+    :param function: The type of the desired node ("protein", "complex", etc)
+    :param variants: A list of variants that the desired node should contain.
+     Note: nodes that contain the variants in addition to specified variants will be matched.
+    :return: A list containing all the nodes that match the given criteria
+    """
+    if not (name or identifier):
+        raise ValueError("Either a 'name' or 'identifier' is required to find a node.")
+    # The mongo .find() method requires that sub-elements be notated 'parent.child'
+    concept_name = '.'.join((CONCEPT, NAME))
+    concept_identifier = '.'.join((CONCEPT, IDENTIFIER))
+    filter_ = {
+        TYPE: NODE,
+        concept_name: name,
+        concept_identifier: identifier,
+        FUNCTION: function,
+        VARIANTS: variants,
+    }
+    for key, value in list(filter_.items()):
+        if not value:
+            del filter_[key]
+
+    return list(collection.find(filter_))
```

Comment:
can you turn these into PyBEL entities using `pybel.tokens.parse_result_to_dsl`? If the nodes are in the right structure, you can do:
```python
return [
    pybel.tokens.parse_result_to_dsl(node)
    for node in collection.find(filter_)
]
```